include/CXXGraph/Graph/Algorithm/Dinic_impl.hpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <queue>
+#include <limits>
+#include <unordered_map>
+#include <vector>
+#include "Node_impl.h"
+#include "CXXGraph/Graph/Graph_decl.h"
+
+
+namespace CXXGraph {
+
+template <typename T>
+class Graph {
+private:
+    std::unordered_map<CXXGraph::id_t, Node<T>> nodes;
```

Comment:
These should not be a class variable, reduce the scope of the variable to the minimum required